src/ReactDOMRe.re

```diff
@@ -124,11 +124,16 @@ module Ref = {
 /** Type-safe form input values. */
 module Input = {
   type t('a);
+  type step('a);
 
   external string: string => t(string) = "%identity";
   external date: Js.Date.t => t(Js.Date.t) = "%identity";
   external float: float => t(float) = "%identity";
   external int: int => t(int) = "%identity";
+
+  external dateStep: float => step(Js.Date.t) = "%identity";
```

Comment:
Good catch! Btw I took a closer look, and according to https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-step , the `step` attribute is always a number. So it would never be a date. I guess we can get rid of `dateStep`!

Reply:
`step` is allowed for dates, but I'm not sure if it can be a float. See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date.

There is apparently also `any` as a valid value for step, I've been thinking of ways to handle that. I'd think of `@bs.inline` but that probably requires `t('a) = string` to work properly. I'll give that a try. I might make `Step` a submodule of `Input`. Any opinions `intStep` ot `Step.int`?

I'd originally included `step: string` to allow an escape hatch. I think we can omit that when the input type is string, but perhaps it's useful to allow people the possibility to pass all these props as a string.

I don't believe this level of safety can be achieved with @bs.unwrap!

Reply:
If we want to support `any` we would need to do a little more trickery, e.g.

```reason
module Input: {
  type step(_);
  ...
  [@bs.inline "any"] let any: step(_);
  external dateStep: float => step(Js.Date.t) = "%identity";
} = {
  type step(_) = string;
  ...
  [@bs.inline] let any = "any";
  external dateStep: float => step(Js.Date.t) = "%identity";
};
```

So we take advantage of JavaScript's dynamism here to allow the `any` string as well as specific ints and floats. `any` seems like it should be the default though, if no `step` is specified. Maybe not worth worrying about.

Re: submodules, I'd say let's keep everything in `Input` for now, we're not at the point where it's overwhelming.

Reply:
Just saw your comment now, so we've converged to the same trick, I'll update the PR and revise `Input` to avoid the submodule.

Reply:
Default value for `step` is 1 for all types of input, according to Mozilla documentation. Since `type step('a) = string` only actually matters for inlined values, keeping `anyStep` shouldn't hurt, but I'm not particularly attached to it either.